tslearn/clustering.py

```diff
@@ -163,16 +174,24 @@ def silhouette_score(X, labels, metric=None, sample_size=None,
     sklearn_metric = None
     if metric_params is None:
         metric_params = {}
+    else:
+        metric_params = metric_params.copy()
```

Comment:
This code is a bit weird (doing a copy of itself): either you create a new object (with a different name), or the copy is unnecessary